### ISSUE TITLE
Create self-compiling example

### ISSUE THREAD

**Issue by Skalman** [pyjs/original-pyjs#789]
It would be cool to have the translator compiling itself. Not sure where this would go, but for example:
- add an option to do this to `pyjs/setup.py`
- provide as an example among the other [examples](http://pyjs.org/examples/)

Preferably this could then be provided as a standalone library.

My usecase would be to compile and run Python code directly in the browser.

**Comment by anthonyrisinger** [comment]
yes, this is a pretty ambitious target... certainly do-able but there is no shortage of issue to tackle beforehand.  this was the subject of a GSoC project 2 years ago, the developer made a fair amount of progress but sort of dropped off the grid, and his work was only partially merged back in.

while this is needed to proclaim "completeness", i've yet to hear any strong use cases to justify its time/complexity, and it seems to amount to little more than a party trick... i don't intend to pursue this, time better spent in 1000x elsewhere IMO.

of course, nothing preventing anyone else from working on it ;) some of the changes ill be making to the core translator/linker (making it pluggable with different strategies) will probably make this more feasible.

**Comment by Skalman** [comment]
That's unfortunate - I was hoping that it'd be easy. I just figured it would be nice to have Python code running in the browser.

Anyway, as the pyjs _does_ become more complete, I guess it'll be an ever easier task. Perhaps in five years it'll self-compile by accident. :-)
